rest-api/dao/metric_set_dao.py

```diff
@@ -1,17 +1,28 @@
-from dao.base_dao import BaseDao
+from dao.base_dao import UpsertableDao
 from model.metric_set import AggregateMetrics, MetricSet
 from dao import database_factory
 
 
-class MetricSetDao(BaseDao):
+class MetricSetDao(UpsertableDao):
 
   def __init__(self):
     super(MetricSetDao, self).__init__(MetricSet,
                                        db=database_factory.get_generic_database())
 
 
-class AggregateMetricsDao(BaseDao):
+class AggregateMetricsDao(UpsertableDao):
 
   def __init__(self):
     super(AggregateMetricsDao, self).__init__(
         AggregateMetrics, db=database_factory.get_generic_database())
+
+  def delete_all_for_metric_set_with_session(self, session, ms_id):
+    aggs = (session.query(AggregateMetrics)
```

Comment:
Rather than doing query followed by delete, I think we should use delete():

http://docs.sqlalchemy.org/en/latest/core/dml.html

Seems more efficient?

Reply:
Thanks, done. This appears to be the right way to use that in combination with the declarative base model.